keps/sig-node/4112-passdown-resources-to-cri/README.md

```diff
@@ -0,0 +1,1302 @@
+
+#### Story 3
+
+As a cluster administrator, I want to install an NRI plugin that does
+customized resource handling. I run kubelet with CPU manager and memory manager
+disabled (CPU manager policy set to `none`). Instead I use my NRI plugin to do
+customized resource allocation (e.g. cpu and memory pinning). To do that
+properly I need the actual resource requests and limits requested by the user.
+
+#### Story 4
+
+As a cluster administrator, I want to install an NRI plugin that does limits
+enforcement for extended resources that the kubelet does not know about (e.g.,
+`misc` cgroup settings). These extended resources are consumed either via the
+pod spec or indirectly via Pod overhead.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The proposal adds new informational data to the CRI API between kubelet and the
+container runtime but there are no user-visible changes. This mitigates
+possible risks considerably.
+
+Data duplication/inconsistency with native resources could be considered a risk
+as those are passed down to CRI both as "raw" requests and limits and as
+"translated" resource control parameters (like cpu shares oom scoring etc). But
+this should be largely mitigated by code reviews and unit tests.
+
+## Design Details
+
+The proposal is that kubelet discloses full resources information from the
+PodSpec to the container runtime. This is accomplished by extending the
+ContainerConfig, UpdateContainerResourcesRequest and PodSandboxConfig messages
+of the CRI API.
+
+However, the proposal does not suggest any changes to the division of
+responsibilities between the kubelet and the runtime or in the pod and
+container lifecycle management. The additional information can be thought seen
+as a "forward lookup" of things that are going to happen in the future.
+Information that can be used in scenarios that depend on it.
+
+With the extended PodSandboxConfig, the runtime can see the full picture of
+Pod's resource usage at Pod creation time, for example enabling more holistic
+resource allocation and thus better interoperability between containers inside
+the Pod. The runtime may use the information to make preparations for all
+upcoming containers of the pod. E.g.  setup all needed resources for a VM-based
+pod or prepare for optimal allocation of resources of all the containers of the
```

Comment:
Can we provide more details around this aspect? It isn't clear to me from reading the KEP how the values will be used for _optimal allocation_. Maybe an example would help shed some light?